code/modules/mob/mob_grab.dm

```diff
@@ -424,17 +424,29 @@
 	return EAT_TIME_FAT //if it doesn't fit into the above, it's probably a fat guy, take EAT_TIME_FAT to do it
 
 /obj/item/weapon/grab/dropped()
-	qdel(src)
+	loc = null
+	if(!destroying)
+		qdel(src)
+
+/obj/item/weapon/grab
+	var/destroying = 0
 
 /obj/item/weapon/grab/Destroy()
+	animate(affecting, pixel_x = 0, pixel_y = 0, 4, 1, LINEAR_EASING)

```

Comment:
I removed this specifically because grabs were not deleting themselves fast enough. It may have been a different issue, however.